tests/std/tests/P0009R18_mdspan_layout_right/test.cpp

```diff
@@ -0,0 +1,390 @@
+        assert(!(m != m));
+        // Other tests are defined in 'check_comparisons' function
+    }
+}
+
+template <class IndexType, size_t... Extents>
+constexpr void check_members(extents<IndexType, Extents...> ext) {
+    do_check_members<IndexType, Extents...>(ext, make_index_sequence<sizeof...(Extents)>{});
+}
+
+constexpr void check_construction_from_other_right_mapping() {
+    { // Check invalid construction
+        using Mapping = layout_right::mapping<extents<int, 3, 3>>;
+        static_assert(!is_constructible_v<Mapping, layout_right::mapping<extents<int, 3>>>);
+        static_assert(!is_constructible_v<Mapping, layout_right::mapping<extents<int, 3, 3, 3>>>);
+    }
+
+    { // Check implicit conversions
+        static_assert(!NotImplicitlyConstructibleFrom<layout_right::mapping<extents<int, 3>>,
+                      layout_right::mapping<extents<int, 3>>>);
+        static_assert(NotImplicitlyConstructibleFrom<layout_right::mapping<extents<int, 3>>,
+            layout_right::mapping<extents<long long, 3>>>);
+        static_assert(NotImplicitlyConstructibleFrom<layout_right::mapping<extents<int, 3, 3>>,
+            layout_right::mapping<extents<int, dynamic_extent, 3>>>);
+        static_assert(NotImplicitlyConstructibleFrom<layout_right::mapping<extents<int, 3, 3>>,
+            layout_right::mapping<extents<int, dynamic_extent, dynamic_extent>>>);
+    }
+}
+
+constexpr void check_construction_from_other_left_mapping() {
+    { // Check construction from layout_left::mapping<E> with various values of E::rank()
+        static_assert(
+            is_constructible_v<layout_right::mapping<dextents<int, 0>>, layout_left::mapping<dextents<int, 0>>>);
+        static_assert(
+            is_constructible_v<layout_right::mapping<dextents<int, 1>>, layout_left::mapping<dextents<int, 1>>>);
+        static_assert(
+            !is_constructible_v<layout_right::mapping<dextents<int, 2>>, layout_left::mapping<dextents<int, 2>>>);
+        static_assert(
+            !is_constructible_v<layout_right::mapping<dextents<int, 3>>, layout_left::mapping<dextents<int, 3>>>);
+    }
+
+    { // Check construction from layout_left::mapping<E> when E is invalid
+        using Mapping = layout_right::mapping<extents<int, 3>>;
+        static_assert(!is_constructible_v<Mapping, layout_left::mapping<extents<int, 1>>>);
+        static_assert(!is_constructible_v<Mapping, layout_left::mapping<extents<int, 2>>>);
+    }
+
+    { // Check implicit conversions
+        static_assert(!NotImplicitlyConstructibleFrom<layout_right::mapping<extents<int, 3>>,
+                      layout_left::mapping<extents<int, 3>>>);
+        static_assert(NotImplicitlyConstructibleFrom<layout_right::mapping<extents<int, 3>>,
+            layout_left::mapping<extents<long long, 3>>>);
+        static_assert(NotImplicitlyConstructibleFrom<layout_right::mapping<extents<int, 3>>,
+            layout_left::mapping<extents<int, dynamic_extent>>>);
+    }

```

Comment:
Suggestion: `layout_right::mapping<extents<int, 3>>` is repeated 3 times here, so it would be clearer to extract a `Mapping` typedef, like the block immediately above. This would also avoid `static_assert` wrapping, making the variation on the RHS even more clear.